collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -374,9 +374,7 @@ private void updateIndex(boolean isAsync, @Nullable FailedValidationResult valid
     public void exit() {
         formSessionRepository.clear(sessionId);
         ReferenceManager.instance().reset();
-        if (form != null) {
```

Comment:
`form` can be `null` if it can't be opened (as in the case described in the issue). In that situation, if a lock was acquired during form opening, it couldn’t be released on form exit.
We no longer need that check, since we now use `FORM_ENTRY_TOKEN` to acquire and release the lock (previously it was `formFilePath`).